crates/js-component-bindgen/src/transpile_bindgen.rs

```diff
@@ -326,8 +349,31 @@ impl<'a> JsBindgen<'a> {
                     {}
                     {}
                     {maybe_init_export}const $init = (async() => {{
```

Comment:
If you're in an async function you don't have to juggle promises manually. You can drive the generator with something like

```js
let done, value, threw;
while (true) {
  0, { done, value } = threw ? gen.throw(value) : gen.next(value);
  if (done) break;
  try {
    value = await value;
    threw = false;
  } catch (e) {
    value = e;
    threw = true;
  }
}
return value;
```

Reply:
For now I've gone with the "hybrid bindgen" implementation, but specializing to sync and async better could make sense.